src/vs/vscode.d.ts

```diff
@@ -5199,6 +5199,32 @@ declare module 'vscode' {
 		export const onDidChangeWorkspaceFolders: Event<WorkspaceFoldersChangeEvent>;
 
 		/**
+		 * Adds workspace folders to the currently opened workspace. Will ignore workspace
+		 * folders that are already part of the workspace.
+		 *
+		 * Note: if this workspace had no folder opened, all extensions will be restarted
+		 * so that the (deprecated) `rootPath` property is updated to point to the first workspace
+		 * folder.
+		 *
+		 * @param folders a list of workspace folders to add.
+		 */
+		export function addWorkspaceFolders(folders: { uri: Uri, name?: string }[]): Thenable<void>;
```

Comment:
Is adding many folders at once really a use-case? I'd expect this to be called for single folders in 99% of the cases. My proposal would be `function addWorkspaceFolder(uri: Uri, name?: string)`

Reply:
Fixed via https://github.com/Microsoft/vscode/pull/36820/commits/158d1520d6823b4584f686f564b2a35ace6d4c55